Bloom.md

```diff
@@ -0,0 +1,167 @@
+
+### Keyspace Event Notification
+
+Every bloom filter based write command will be made to publish a keyspace event after the data is mutated.
+* Event type: VALKEYMODULE_NOTIFY_GENERIC
+* Event name: command name in lowercase. e.g., BF.ADD command publishes event "bloom.add".
+
+Users can subscribe to the bloom events via the standard keyspace event pub/sub. For example,
+
+```text
+1. enable keyspace event notifications:
+    valkey-cli config set notify-keyspace-events KEA
+2. suscribe to keyspace & keyevent event channels:
+    valkey-cli psubscribe '__key*__:*'
+```
+
+### Scalale filters / Non Scaling filters
+
+Bloom Filters can either be configured as scalable (default) or non-scalable.
+
+### Choice of Bloom Filter Library
+
+We evaluated the following libraries:
+* https://crates.io/crates/bloomfilter
+* https://crates.io/crates/growable-bloom-filter
+* https://crates.io/crates/fastbloom
+
+Here are some important factors when evaluating libraries:
+* Is it actively maintained?
+* Has there been any breaking changes in serialization and deserialization of bloom filters over version updates?
+* Performance
+* Popularity
+
+https://crates.io/crates/bloomfilter was chosen as it provides all the required APIs for controlling a bloom filter. This library has also been stable through the releases.
+
+Certain libraries (e.g. fastbloom) are no longer compatible with older versions (after updates) resuling in serialization / deserialization incompatibility. 
+
+
+### Max number of filters for scalable bloom filters
+
+### Large BloomFilter objects
+
+Create and Delete operations on large bloom filters take longer durations and will block the main thread for this duration. 
+Because of this, the following operations can be handled differently:
+* defrag callback: If the memory used is greater than X MB (configurable with `bf.bloom_large_item_threshold`), we can skip defrag operations on this bloom object.
+* free_effort callback: This callback decides the free effort for the bloom object. If it is greater than X MB (configurable with `bf.bloom_large_item_threshold`), we can return 0 to use async free logic.
+* create operations (BF.ADD/MADD/INSERT): We can compute the estimated size of the bloom filter that will be created as a result of the operation. We can consider rejecting requests in case of unavailable memory.
+
+
+## Specification
+
+### Bloom Filter Command API
+
+The following Bloom Filter commands are supported and their API syntax is compatible with ReBloom:
+BF.ADD <key> <item>
+BF.EXISTS <key> <item>
+BF.MADD <key> <item> [<item> ...]
+BF.MEXISTS <key> <item> [<item> ...]
+BF.CARD <key>
+BF.INFO <key> [CAPACITY | SIZE | FILTERS | ITEMS | EXPANSION]
+BF.RESERVE <key> <false_positive_rate> <capacity> [EXPANSION <expansion>] | [NONSCALING]
+BF.INSERT <key> [ERROR <fp_error>] [CAPACITY <capacity>] [EXPANSION <expansion>] [NOCREATE] [NONSCALING] ITEMS <item> [<item> ...]
```

Comment:
Can you double check if the ValkeyBloom has same command syntax with Redis Bloom，because ReBloom syntax is: 
 BF.INSERT key [CAPACITY capacity] [ERROR error]
  [EXPANSION expansion] [NOCREATE] [NONSCALING] ITEMS item [item
  ...]  https://redis.io/docs/latest/commands/bf.insert/

Reply:
ReBloom's BF.INSERT command accepts any order of these optional arguments.

ValkeyBloom also does the same.

I can update the documentation so that it does not cause any confusion regarding this